### DISCUSSION TITLE
when to call backupOneMMKV and restoreMMKV in ios?

### DISCUSSION THREAD

**Discussion by BannaGo** [Tencent/MMKV#1078] [General]
在ios中，我们应该何时调用backup: 和restore:的接口了? 
如果我没有主动调用backupOneMMKV, 那么数据恢复功能能正常工作吗？它又是从哪里来恢复数据了？
我注意到，在数据恢复中，有
- (MMKVRecoverStrategic)onMMKVCRCCheckFail:(NSString *)mmapID {
    return MMKVOnErrorRecover;
}

- (MMKVRecoverStrategic)onMMKVFileLengthError:(NSString *)mmapID {
    return MMKVOnErrorRecover;
}

如果需要我主动调用backupOneMMKV和restore...，那我分别应该在什么时机调用了？

感谢您的解答

**Comment by lingol** [comment]
Simple answer: the backup and restore API is not for corruption protection. It's for data migration such as export & import.
